packages/material/src/controls/MaterialBooleanControl.tsx

```diff
@@ -24,47 +24,47 @@
 */
 import isEmpty from 'lodash/isEmpty';
 import React from 'react';
-import { connect } from 'react-redux';
 import {
-  ControlProps,
   isBooleanControl,
-  mapDispatchToControlProps,
-  mapStateToControlProps,
+  OwnPropsOfControl,
   RankedTester,
   rankWith,
 } from '@jsonforms/core';
-
+import { ctxDispatchToControlProps, ctxToControlProps, useJsonForms } from '@jsonforms/react';
 import { FormControlLabel, Hidden } from '@material-ui/core';
 import { MuiCheckbox } from '../mui-controls/MuiCheckbox';
 
-export const MaterialBooleanControl = ({
-  label,
-  id,
-  visible,
-  ...props
-}: ControlProps) => (
-  <Hidden xsUp={!visible}>
-    <FormControlLabel
-      label={label}
-      id={id}
-      control={
-        <MuiCheckbox
-          id={id + '-input'}
-          isValid={isEmpty(props.errors)}
-          visible={visible}
-          {...props}
-        />
-      }
-    />
-  </Hidden>
-);
+export const MaterialBooleanControl = (props: OwnPropsOfControl) => {
+  const ctx = useJsonForms();
+  const { data, enabled, uischema, path, visible, label, id, errors, schema, rootSchema } = ctxToControlProps(ctx, props);
```

Comment:
Maybe use a HigherOrderComponent to not duplicate code and have less changes on the renderers?